lib/utils/helpers.ts

```diff
@@ -96,9 +97,26 @@ export async function sudo(
 }
 
 export function runCommand<T>(command: string): Promise<T> {
-	const capitano = require('capitano') as typeof import('capitano');
-	const capitanoRunAsync = promisify(capitano.run);
-	return capitanoRunAsync(command);
+	let commandArgs = command.split(' ');
```

Comment:
Splitting command lines with spaces is problematic. For example, above you mentioned _"single quotes needed to be removed around the paths here."_ In a command line like `os configure ${path} --device ${uuid}`, the `path` could contain blank spaces, and `command.split(' ')` would split a single path as multiple separate arguments even if the path is quoted (and different shells like bash / cmd.exe / powershell have different quoting rules).

Anyway, my suggestion is to change the `runCommand` function interface to take an array of strings instead of a single string, thus avoiding the need for `command.split()` here:
```
export function runCommand<T>(commandArgs: string[]): Promise<T> {
```
Then in the caller, instead of:
```
	async configureOsImage(path: string, uuid: string, options: FlagsDef) {
		let configureCommand = `os configure ${path} --device ${uuid}`;
		...
		await runCommand(configureCommand);
```
We'd have:
```
	async configureOsImage(path: string, uuid: string, options: FlagsDef) {
		let configureCommand = ['os', 'configure', path, '--device', uuid];
		...
		await runCommand(configureCommand);
```
Unless I'm missing something, I feel that using arrays all along is preferable to using `string.split(' ')`.

By the way, even better would be to avoid `runCommand` altogether -- to avoid "command-command calls" like you described it. Instead of `balena device init` calling `balena os configure`, it should just call a Javascript function that does what `balena os configure` does without having to parse another command line and go multiple times through validation checks, whoami / logged in, mixpanel (?) and all. But this would be a more involved refactoring; perhaps one step at a time.

Reply:
Yeah I think the best is to get rid of this entirely, and call isolated logic.   (BTW, mixpanel and probably some of the others you mentioned are not invoked my this).